jwst/assign_wcs/niriss.py

```diff
@@ -108,12 +110,20 @@ def niriss_soss(input_model, reference_files):
 
     Notes
     -----
-    It includes tWO coordinate frames -
+    It includes two coordinate frames -
     "detector" and "world".
 
-    It uses the "specwcs" reference file.
+    It uses the "specwcs" reference file type.
     """
 
+    # The input is the grism image
+    if not isinstance(input_model, ImageModel):
```

Comment:
SOSS mode data are usually taken as a time-series observation, which are always processed on a per-integration basis, and hence the input to the spec2 and spec3 pipeline levels is expected to be a `CubeModel` (3-D stack of integrations), not an `ImageModel`. Unfortunately, it's not mandatory that SOSS uses the time-series approach, so the input could be either `ImageModel` (2D) or `CubeModel` (3D). You never know.